### PR TITLE
Settings window navigation tree icons

### PR DESCRIPTION
**Pull request by NikolayXHD**
Before: 

image

After:

image

### CODE REVIEW

**Comment by pmiossec** [comment]
Great!

Perhaps you could also add ones for the plug-ins. The ones already used in the "Browse Form" menu..

**Comment by NikolayXHD** [comment]
> Where are the new icons from?
I painted the new icons, except from ones copied from `GitUI\Resources\Icons\originals` andGit logo

> Perhaps you could also add ones for the plug-ins. The ones already used in the "Browse Form" menu..
Ok, I will

**Comment by RussKie** [comment]
Nice work!
Some (if not most) icons in v3 comes from https://github.com/yusukekamiyamane/fugue-icons, they come with CCA 3.0 lic, so you can use them.

For colours a something like "palette" would work, also there are plenty of icons for "font" methaphor.

**Comment by NikolayXHD** [comment]
Thanks for tip @RussKie!
Replaced appearance icon lipstick -> palette

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs [hunk 1]
```diff
@@ -67,6 +69,35 @@ public void AddSettingsPage(ISettingsPage page, SettingsPageReference parentPage
...
+        private int GetImageIndex(Bitmap icon)
+        {
+            if (icon == null)
+            {
+                // 0 -> Images.Blank
+                // not -1 because because TreeView displays ImageList[0] when ImageIndex == -1
+                return 0;
+            }
+
+            int imageIndex = treeView1.ImageList.Images.Cast<Bitmap>()
```
Comment (RussKie) [inline GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: is this check really needed? this method is called for each node once when the dialog is created.

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs [hunk 2]
```diff
@@ -32,17 +33,18 @@ public SettingsTreeViewUserControl()
 
             treeView1.ImageList = new ImageList();
             treeView1.ImageList.ImageSize = DpiUtil.Scale(new Size(16, 16)); // Scale ImageSize and images scale automatically
+            treeView1.ImageList.Images.Add(Images.Blank);
             this.AdjustForDpiScaling();
         }
 
         /// <param name="asRoot">only one page can be set as the root page (for the GitExt and Plugin root node)</param>
-        public void AddSettingsPage(ISettingsPage page, SettingsPageReference parentPageReference, bool asRoot = false)
+        public void AddSettingsPage(ISettingsPage page, SettingsPageReference parentPageReference, Bitmap icon = null, bool asRoot = false)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: I would make `Bitmap icon` a mandatory parameter, the two (?) instances that don't have images can explicitly supply `null`

Comment (RussKie) [inline GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Can we use more generic type `Image` instead `Bitmap`?

**Comment by RussKie** [comment]
As an unrelated observation - we need to order the plugins in A-Z order

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs (hunk 1 above)
Comment (NikolayXHD) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Actually I see you are right. Even though some tree nodes have the same Icon, there is no harm in duplicating them in ImageList, as we are only duplicating the reference to a bitmap, not the bitmap itself.

Will be fixed.

File: GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs (hunk 2 above)
Comment (NikolayXHD) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Will be fixed

Comment (NikolayXHD) [reply GitUI/CommandsDialogs/SettingsDialog/SettingsTreeViewUserControl.cs]: Will be fixed

**Comment by NikolayXHD** [comment]
> we need to order the plugins in A-Z order

Will be fixed

**Comment by NikolayXHD** [comment]
Done

**Pull request merged**